python/src/scanner.rs

```diff
@@ -28,6 +30,129 @@ use pyo3::exceptions::PyValueError;
 use crate::reader::LanceReader;
 use crate::RT;
 
+#[pyclass]
+pub struct LanceScanStats {
+    inner: ScannerStats,
+}
+
+impl LanceScanStats {
+    pub fn new(inner: ScannerStats) -> Self {
+        Self { inner }
+    }
+}
+
+#[pymethods]
+impl LanceScanStats {
+    /// The start of the scan
+    ///
+    /// This is when the stream is constructed, not when it is first consumed.
+    ///
+    /// Returned as milliseconds since the UNIX epoch
+    #[getter]
+    fn start(&self) -> PyResult<u64> {
+        Ok(self
+            .inner
+            .start
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap()
+            .as_millis()
+            .try_into()
+            .unwrap())
+    }
+
+    /// The end of the scan
+    ///
+    /// This is when the last batch is provided to the consumer which may be
+    /// well after the I/O has finished (if there is a slow consumer or expensive
+    /// decode).
+    ///
+    /// Returned as milliseconds since the UNIX epoch
+    #[getter]
+    fn end(&self) -> PyResult<u64> {
+        Ok(self
+            .inner
+            .end
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap()
+            .as_millis()
+            .try_into()
+            .unwrap())
+    }
+
+    /// The wall clock duration of the scan
+    ///
+    /// NOTE: This is not the time that the scanner was actually doing work, and not the amount
+    /// of time spent in I/O but simply the time from when the scanner was created to when the
+    /// last batch was provided to the consumer.
+    ///
+    /// As an example, if a consumer is slow to consume the data (e.g. they are writing the data
+    /// back out to disk or doing expensive processing) then this will be much larger than the
+    /// actual time to read the data.
+    ///
+    /// Returned as floating point seconds
+    #[getter]
+    fn wall_clock_duration(&self) -> PyResult<f64> {
+        Ok(self.inner.wall_clock_duration.as_secs_f64())
+    }
```

Comment:
Same thing here with `Duration`. I think it can natively convert to Python's `timedelta`.